aws/src/main/java/org/apache/iceberg/aws/dynamodb/DynamoDbTableOperations.java

```diff
@@ -135,8 +135,7 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
           io().deleteFile(newMetadataLocation);
         }
       } catch (RuntimeException e) {
-        LOG.error("Fail to cleanup metadata file at {}", newMetadataLocation, e);
-        throw e;
+        LOG.error("Failed to cleanup metadata file at {}", newMetadataLocation, e);
```

Comment:
Would you consider adding `e` as a "suppressed" exception to any previous exception?

It's a bit more refactoring, but often having more information in the thrown exception is more useful than logs.

Reply:
I think this is fine. If it were easier, I'd opt to suppress the exception (maybe we should consider using `runSafely` for that) but since this is just warning that a file wasn't cleaned up, the important thing is not to throw.